Assignment 2.Rmd

```diff
@@ -67,33 +77,175 @@ library(igraph)
 
-Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+
+```{r}
+library(tidyverse)
+rm(list = ls())
+#read in data.
+data = read.csv(file = "hudk4050-classes.csv")
+
+#combine name to be full 
+data <- data %>% unite(full_name, First.Name, Last.Name, sep="_")
+
+#remove spaces in names column 
+data$full_name <- (gsub(" ", "", data$full_name))
+
+data2 <- gather(data, class_type, course_name, 2:7)
+
+#remove class_type
+data2$class_type <- NULL
+
+#remove spaces from course names 
+data2$course_name <- (gsub(" ", "", data2$course_name))
+
+#replace tricky entries
+data2$course_name <- (gsub("HUDK4050", "4050", data2$course_name))
+data2$course_name <- (gsub("4050", "HUDK4050", data2$course_name))
+data2$course_name <- (gsub("HUDMHUDK4050", "HUDM4050", data2$course_name))
+
+data2$course_name <- (gsub("HUDM4125", "4125", data2$course_name))
+data2$course_name <- (gsub("4125", "HUDM4125", data2$course_name))
+
+data2$course_name <- (gsub("HUDM5026", "5026", data2$course_name))
+data2$course_name <- (gsub("5026", "HUDM5026", data2$course_name))
+
+data2$course_name <- (gsub("HUDM5126", "5126", data2$course_name))
+data2$course_name <- (gsub("5126", "HUDM5126", data2$course_name))
+
+#remove duplicates
+data2 <- unique(data2)
+
+#remove HUDK4050 class for everyone
+data2 <-data2[(data2$course_name!="HUDK4050"),]
+
+#add column of 1s
+data2$count <- "1"
+
+#create new matrix how we want it
+data3 <- spread(data2, course_name, count)
+
+#remove column of V1 (blank)
+data3$V1 <- NULL
+
+#replace NA with zero
+data3[is.na(data3)] <- 0
+```
+
+
+```{r}
+#move full name to be row name
+row.names(data3) <- data3$full_name
+data3$full_name <- NULL
+
+#add ID number to name list so we can match them later 
+data$id <- seq(1,53,1)
+
+#trying to multiply matrices to get matrix to plot 
+data3 <- as.data.frame(sapply(data3, as.numeric))
+data3 <- as.matrix((data3))
+data4 <- t(data3)
+
+data5 <- data3 %*% data4
+
+#add back names
+names <- data[,c("id", "full_name")]
+
+row.names(data5) <- names$full_name
+colnames(data5) <- names$full_name
+```
+
+plot the graph from the adjacency matrix 
+```{r}
+library(igraph)
+
+graph2 <- graph_from_adjacency_matrix(data5, weighted = TRUE,
+  diag = FALSE, add.colnames = NULL, add.rownames = NA)
+
+#plot graph
+plot(graph2,layout=layout.fruchterman.reingold,edge.arrow.size=0.01)
+
+```
+
+
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and degree. **Who is the most central person in the network?**
+
+```{r}
+
+#degree
+#number of connections you have
+which.max(degree(graph2))
+#Eudora Xinyi Niu 
+
+
+#betweenness centrality
+#bottleneck in network. Very connected
+which.max(betweenness(graph2))
+#Eudora Xinyi Niu (again)
+
+#Conclusion: Eudora Xinyi Niu is most central person in the network!
```

Comment:
Try to interpret this within the context of this dataset. As in, what does it mean to be the most central person in this network?